src/app-locator.js

```diff
@@ -0,0 +1,98 @@
+import appIds from './known-app-ids'
+
+// Known apps and their assigned ports
+const DEFAULT_LOCAL_URLS = {
+  [appIds['Voting']]: 'http://localhost:3001/',
+  [appIds['Finance']]: 'http://localhost:3002/',
+  [appIds['TokenManager']]: 'http://localhost:3003/',
+  [appIds['Survey']]: 'http://localhost:3004/',
+  [appIds['Agent']]: 'http://localhost:3005/',
+  [appIds['Fundraising']]: 'http://localhost:3006/',
+}
+
+// Split a single bridge declaration into an app + a location
+function splitBridge(bridge) {
+  const firstColon = bridge.indexOf(':')
+  if (firstColon === -1) {
+    return [bridge]
+  }
+  return [
+    bridge.slice(0, firstColon).trim(),
+    bridge.slice(firstColon + 1).trim(),
+  ]
+}
+
+function getAppId(app) {
+  // known app name
+  if (appIds[app]) {
+    return appIds[app]
+  }
+
+  // probably a valid app ID
+  if (app.startsWith('0x') && app.length === 66) {
+    return app
+  }
+
+  return null
+}
+
+function getAppUrl(appId, location) {
+  // get the default URL for this appId (if location is local)
+  if ((!location || location === 'local') && DEFAULT_LOCAL_URLS[appId]) {
+    return DEFAULT_LOCAL_URLS[appId]
+  }
+
```

Comment:
We should either default `location` to `''`, or guard this section to make sure `location` is a string; right now if you give a non-defaulted appId and null or undefined `location`, this will break.

Reply:
Done in https://github.com/aragon/aragon/pull/1124/commits/1503a8c3962cbfe8f39b033841d9b0aaf3314176 but I don’t think it could be anything else than a string? It’s good to have it though, as it shows the expected type.